bazel/upload_systest_dep.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+#
+# Upload a dependency to shared storage and returns the download URL.
+#
+# The path to the dependency should be specified as the first (and only) argument.
+#
+# The download URL is printed to stdout.
+
+# NOTE: This script uses bazel-remote as the CAS storage (implementation detail).
+
+# Look up a CAS key (provided as $1) through the redirect server.
+# If the key exists, then then download URL is returned (through stdout).
+# If the key does not exist, the empty string is returned.
+lookup_dep_url() {
+    REDIRECT_SERVER_URL="https://artifacts.idx.dfinity.network"
+    local redirect_url="$REDIRECT_SERVER_URL/cas/$1"
+    local result
+    result=$(curl --silent --head \
+        -w '%{http_code} %{redirect_url}' \
+        "$redirect_url" \
+        | tail -n1)
+
+    local result_code
+    result_code=$(cut -d' ' -f1 <<<"$result")
+    if [ "$result_code" == "404" ]; then
+        # The key was not found
+        return
+    fi
+
+    if [ "$result_code" != "307" ]; then
+        echo "Expected 404 or 307 when looking up dependency '$1', got '$result_code'" >&2
+        exit 1
+    fi
+
+    local result_url
+    result_url=$(cut -d' ' -f2 <<<"$result")
+    if [ -z "$result_url" ]; then
+        echo "Looking up dependency '$1' did not return a URL, got: '$result'" >&2
+        exit 1
+    fi
+
+    echo "$result_url"
+}
+
+dep="${1:?Dependency not specified}"
+dep_filename=${dep#*:}
+
+echo "Found dep to upload $dep_filename ($dep_sha256)" >&2
+result_url=$(lookup_dep_url "$dep_sha256")
+
+# First, figure out _if_ the dep should be uploaded (no point re-uploading several GBs
+# if it's been uploaded already)
+if [ -n "$result_url" ]; then
+    echo "dep '$dep_filename': already uploaded" >&2
+else
+    echo "dep '$dep_filename': not uploaded yet, uploading to $dep_upload_url" >&2
+
+    # Here we figure out the best URL where to upload the dependency. We use the hash of the
+    # empty blob, which will always be found by bazel-remote (baked in) and expect the redirect
+    # server to return the best URL (in practice, the fastest to complete the request)
+    EMPTY_SHA=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+    empty_sha_url=$(lookup_dep_url "$EMPTY_SHA")
+    UPLOAD_URL="${empty_sha_url%"/$EMPTY_SHA"}" # strip the empty blob key from the url
+    echo "Using upload URL: '$UPLOAD_URL'" >&2
+
```

Comment:
Instead of looking up the empty blob through the redirect server and hoping you'll hit the local cluster's bazel-remote why not upload directly to the local cluster?

You can achieve this by uploading to `bazel-remote.idx.dfinity.network` which always [points](https://github.com/dfinity-ops/k8s/blob/main/clusters/zh1-idx1/apps/ingress-bazel-remote.yaml#L6C71-L6C103) to the bazel-remote in the cluster where this script is running.

This ensures you always upload to the local cluster and never cross clusters. Then afterwards you execute:
```sh
result_url=$(lookup_dep_url "$dep_sha256")
```
as you do above to get the final URL which I see you execute anyway below so you can also simplify things this way.

Reply:
Sure thing! I used to push to `http://server.bazel-remote.svc.cluster.local:8080/` which is also the cluster-local bazel-remote, but then realized I might as well look up whatever server is the fastest (and potentially: least busy). Don't care too strongly so will include your suggestion tomorrow!